### DISCUSSION TITLE
Allow users to construct values of the `Nullable` type in const contexts.

### DISCUSSION THREAD

**Discussion by johnchildren** [diesel-rs/diesel#3227] [Ideas]
When using the `AllColumns` pattern described in the ["Composing Applications" guide](https://diesel.rs/guides/composing-applications.html), it is sometimes useful to represent the columns resulting from a left or right join, for example:

```rust
const ALL_FOO: AllFoo = (foo::id, foo::name, foo::bar_id); // bar_id is Nullable
const ALL_BAR: AllBar = (bar::id, bar::name);

type AllFooBar = (
  foo::id,
  foo::name,
  Nullable<AllBar>,
);
const ALL_FOO_BAR: AllFoorBar = (
  foo::id,
  foo::name,
  ALL_BAR.nullable(), // not possible to call in a const context
);
```

Unfortunately as `.nullable()` is a trait method it's not really possible to write `ALL_FOO_BAR` as far as I can tell, but there might be some other workaround?

I think if the `new` constructor for `Nullable` was part of the public API, that would be possible, but I understand it's possible this could be misused? I'm not really sure what a sensible alternative would be however.

**Comment by weiznich** [comment]
This patterns has been replaced by the "new" `Selectable` trait on the upcoming 2.0 release. Checkout [this documentation](https://docs.diesel.rs/master/diesel/expression/trait.Selectable.html) for details.

**Comment by johnchildren** [comment]
Ah thank you! This solves my use case and cuts out a lot of my boilerplate!
